### ISSUE TITLE
Kernel Error when running `fsleyes -nb`

### ISSUE THREAD

**Issue by valfrancois** [pauldmccarthy/fsleyes#62]
Hi, 

I'm trying to interact with fsleyes from a jupyter notebook but I get a red "Kernel error" message when I try one of these ways to open my notebook:
- `fsleyes -S -nbf <my_notebook.ipynb>`
- `fsleyes -nb` -> navigate to `<my_notebook.ipynb>` and open it

The details are:
```
Failed to start the kernel
Unhandled error

Traceback (most recent call last):
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/web.py", line 1704, in _execute
    result = await result
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 769, in run
    yielded = self.gen.throw(*exc_info)  # type: ignore
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/notebook/services/sessions/handlers.py", line 72, in post
    type=mtype))
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 762, in run
    value = future.result()
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 769, in run
    yielded = self.gen.throw(*exc_info)  # type: ignore
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/notebook/services/sessions/sessionmanager.py", line 88, in create_session
    kernel_id = yield self.start_kernel_for_session(session_id, path, name, type, kernel_name)
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 762, in run
    value = future.result()
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 769, in run
    yielded = self.gen.throw(*exc_info)  # type: ignore
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/notebook/services/sessions/sessionmanager.py", line 101, in start_kernel_for_session
    self.kernel_manager.start_kernel(path=kernel_path, kernel_name=kernel_name)
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 762, in run
    value = future.result()
  File "/home/vfrancois/anaconda3/envs/conda3.7/lib/python3.7/site-packages/tornado/gen.py", line 216, in wrapper
    result = ctx_run(func, *args, **kwargs)
  File "/tmp/fsleyes-jupytertzm0601v/config/fsleyes_kernelmanager.py", line 49, in start_kernel
    .start_kernel(**kwargs).result()
AttributeError: 'coroutine' object has no attribute 'result'
```

I installed fsleyes and my notebook dependencies in the conda env `'conda3.7' ` and added it to my Jupyter kernels.

From a quick look at similar issues on stackoverflow it looks like some line wouldn't properly await some result. Not sure if this comes from my install/config or from fsleyes itself as my colleague who has a very similar config doesn't encounter the issue.

OS: ubuntu 18.04
Python version: 3.7.6
```
fsleyes                   0.34.2
fsleyes-props             1.7.0
fsleyes-widgets           0.9.0

```

**Comment by pauldmccarthy** [comment]
I don't have a lot of time to play around right now, but my guess is that this is due to changes in the `notebook` project, migrating to use `coroutines`. Do things work if you set up an environment with versions of packages listed in `requirements-notebook.txt`?

**Comment by valfrancois** [comment]
Ok so I installed the minimum versions in `requirements-notebook.txt`, and the kernel seems to be running correctly now even if I got the following error at fsleyes startup:
```
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/gl/__init__.py", line 636, in create
    ready()
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/main.py", line 580, in realCallback
    callback()
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/main.py", line 371, in buildGui
    frame.menuActions[NotebookAction](namespace[0].notebookFile)
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/actions/base.py", line 172, in __call__
    return self.__func(*args, **kwargs)
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/actions/notebook.py", line 149, in __openNotebooks
    self.__server = self.__startServer(progdlg, nbfile)
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/actions/notebook.py", line 204, in __startServer
    while elapsed < 5 and not server.ready:
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/actions/notebook.py", line 536, in ready
    return self.__readPort() is not None
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/fsleyes/actions/notebook.py", line 525, in __readPort
    for server in notebookapp.list_running_servers():
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/site-packages/notebook/notebookapp.py", line 1701, in list_running_servers
    info = json.load(f)
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/json/__init__.py", line 296, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/vfrancois/anaconda3/envs/test/lib/python3.7/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 2 column 1 (char 1)
```
The notebook correctly interacts with fsleyes after I close the error window so I can live with this error. Thanks!

**Comment by valfrancois** [comment]
Update: upgrading the packages to the following versions got rid of the second error.
```
ipykernel                 5.1.4
ipython                   7.12.0
jupyter_client            5.3.4 
notebook                  6.0.3 
pyzmq                     18.1.1
tornado                   6.0.3 
```

**Comment by pauldmccarthy** [comment]
Great, thanks for the report - I'll fix this when I get a chance

**Comment by jdoepfert** [comment]
+1 for this. Seems to be mainly a problem on Linux/Ubuntu

**Comment by pauldmccarthy** [comment]
This should be fixed in the latest version of FSLeyes (1.0.3) - please let me know if you're still having trouble!
